File: backend/app/schemas/playlist.py
from pydantic import BaseModel
from datetime import datetime

from .playlist_song import PlaylistSongOut

class PlaylistOut(BaseModel):
    id: int
    user_id: int
    name: str
    created_at: datetime | None = None
    genre: str | None = None
    bpm_from: int | None = None
    bpm_to: int | None = None
    description: str | None = None

    playlist_songs: list[PlaylistSongOut]

    class Config:
        from_attributes = True   # IMPORTANT for Pydantic v2

File: frontend/src/types/playlist.ts
export type Playlist = {
  id: number;
  name: string;
  description?: string;
  genre: string | null;      // or `string` if always present
  bpm_from: number | null;
  bpm_to: number | null;
  created_at: string;        // ISO timestamp from API
};